docs/dev_guide/diagnostic_ref/dpct1045.rst

```diff
@@ -100,12 +102,15 @@ which is rewritten to:
                              std::shared_ptr<dpct::sparse::matrix_info> descrA,
                              float alpha, float beta, float *a_val, int *a_row_ptr,
                              int *a_col_ind, float *b, float *c) {
-      a_val = new_a_val;
-      a_row_ptr = new_a_row_ptr;
-      a_col_ind = new_a_col_ind;
       descrA->set_matrix_type(dpct::sparse::matrix_info::matrix_type::ge);
+      float* new_a_val = sycl::malloc_shared<float>(7, *handle);

```

Comment:
```suggestion
      float* new_a_val =allocate_memory(size) // compute new_a_val size from a_val, just as new_a_val in comment

```

Reply:
Maybe we can provide helper function to cal the size of memory required and fill the memory based on a_val.